File: BaseLibrary/command/tank/ShootCommand.cs
using System;
using System.Collections.Generic;
using BaseLibrary.protocol;

namespace BaseLibrary.command.tank {
    public class ShootCommand : ATankCommand {

        private static readonly List<ISubCommandFactory> SUB_COMMAND_FACTORIES = new List<ISubCommandFactory>();

        public static int RegisterSubCommandFactory(ISubCommandFactory subCommandFactory) {
            int position = SUB_COMMAND_FACTORIES.Count;
            SUB_COMMAND_FACTORIES.Add(subCommandFactory);
            return position;
        }

        private double _angle;

        /// <summary>
        /// In witch direction want to shot.
        /// </summary>
        public double ANGLE {
            get {
                if (pending)
                    throw new NotSupportedException("Cannot access to property of pending request.");
                return _angle;
            }
            private set => _angle = value;
        }

        private double _range;
        /// <summary>
        /// How far from want to shot.
        /// </summary>
        public double RANGE {
            get {
                if (pending)
                    throw new NotSupportedException("Cannot access to property of pending request.");
                return _range;
            }
            private set => _range = value;
        }

        public ShootCommand(double range, double angle)
            : base() {
                ANGLE = angle;
                RANGE = range;
            pending = false;
        }
    }
}

File: Java-Client/RobotBattlefield-Client-Java/src/BaseLibrary/communication/command/tank/ShootCommand.java

package BaseLibrary.communication.command.tank;

import BaseLibrary.communication.protocol.AProtocol;

public class ShootCommand extends ATankCommand {
	
	private static final String NAME = "SHOOT";

	static {
		AProtocol.registerForDeserialize(NAME, ShootCommand.class);
	}

	@Override
	public String getCommandName() {
		return NAME;
	}

	/**
	 * How far from want to shot.
	 */
	private double range;
	
	/**
	 * In witch direction want to shot.
	 */
	private double angle;

	

	public ShootCommand(double range, double angle) {
		this.angle = angle;
		this.range = range;
		pending = false;
	}
	
	public double getAngle() {
		if (pending)
			throw new UnsupportedOperationException("Cannot access to property of pending request.");
		return angle;
	}
	
	public double getRange() {
		if (pending)
			throw new UnsupportedOperationException("Cannot access to property of pending request.");
		return range;
	}
}
